components/emoji/render_emoji.tsx

```diff
@@ -1,32 +1,54 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 import React from 'react';
-import {useSelector} from 'react-redux';
+import {useDispatch, useSelector} from 'react-redux';
 
 import {getEmojiImageUrl} from 'mattermost-redux/utils/emoji_utils';
 
+import {getCurrentUserId} from 'mattermost-redux/selectors/entities/users';
+
 import {getEmojiMap} from 'selectors/emojis';
 import {GlobalState} from 'types/store';
+import {ModalIdentifiers} from 'utils/constants';
+import CustomStatusModal from 'components/custom_status/custom_status_modal';
+import {openModal} from 'actions/views/modals';
 
 interface ComponentProps {
     emoji: string;
     size?: number;
     emojiStyle?: React.CSSProperties;
+    openModalOnClick?: boolean;
 }
 
-const RenderEmoji = ({emoji, emojiStyle, size}: ComponentProps) => {
+const RenderEmoji = ({emoji, emojiStyle, size, openModalOnClick}: ComponentProps) => {
     if (!emoji) {
         return null;
     }
 
+    const dispatch = useDispatch();
     const emojiMap = useSelector((state: GlobalState) => getEmojiMap(state));
+    const currentUserId = useSelector((state: GlobalState) => getCurrentUserId(state));
     const emojiFromMap = emojiMap.get(emoji);
     if (!emojiFromMap) {
         return null;
     }
     const emojiImageUrl = getEmojiImageUrl(emojiFromMap);
+
+    const openCustomStatusModal = (event: React.MouseEvent<HTMLSpanElement, MouseEvent>) => {
+        if (openModalOnClick) {
+            event.stopPropagation();
+            const customStatusInputModalData = {
+                ModalId: ModalIdentifiers.CUSTOM_STATUS,
+                dialogType: CustomStatusModal,
+                dialogProps: {userId: currentUserId},
+            };
+
+            dispatch(openModal(customStatusInputModalData));
+        }
+    };
```

Comment:
Please pass this function as an `onClick` method from the component that wants to open the custom status modal.
nit - you can also use an action in `mattermost-webapp` for the modal data.